src/Main.purs

```diff
@@ -62,19 +75,31 @@ ui =
           [ HH.text "Robust plot" ]
       , HH.button
           [ HE.onClick \_ -> Just BasicPlot ]
-          [ HH.text "Plot polygon async" ]
+          [ HH.text "Plot example function" ]
+      , HH.button
+          [ HE.onClick \_ -> Just Clear ]
+          [ HH.text "Clear plot" ]
       , HH.slot _canvas 1 (canvasComponent canvasController) state.input absurd
       ]
 
-computePlot :: Plot -> ReaderT Config Aff (DrawCommand Unit)
-computePlot plot = lift $ computePlotAsync plot
+computePlot :: Size -> PlotCommand -> ReaderT Config Aff (DrawCommand Unit)
+computePlot canvasSize plot = lift $ computePlotAsync canvasSize plot 
 
 handleAction :: forall o. Action -> H.HalogenM State Action ChildSlots o (ReaderT Config Aff) Unit
-handleAction = case _ of
-  BasicPlot -> do
-    state <- H.get
-    plotCommands <- lift $ computePlot $ basicPolygon $ xyBounds (-1.0) (1.0) (-1.0) (1.0)
-    H.put state { input { operations = plotCommands } }
+handleAction action = do
+  state <- H.get
+  case action of
+    Init -> do
+      plotCommands <- lift $ computePlot state.input.size $ clear $ state.bounds
+      H.put state { input { operations = plotCommands } }
+    Clear -> do
+      plotCommands <- if state.plotExists 
+        then lift $ computePlot state.input.size $ clear $ state.bounds
+        else pure $ pure unit
+      H.put state { input { operations = plotCommands }, plotExists = false }
+    BasicPlot -> do
+      plotCommands <- lift $ computePlot state.input.size $ basicPlot state.plotExists state.bounds
+      H.put state { input { operations = plotCommands }, plotExists = true }
```

Comment:
I do not quite understand how the asynchronous comptutation impacts this code.  I hope the computation does not block the thread within handleAction.  Somehow I imagine that the asynchronous code, when finished, should trigger its own action.  I would like to clarify this and potentially refactor this soon in future iterations.